admin-dev/themes/new-theme/template/layout.tpl

```diff
@@ -5,7 +5,7 @@
 </head>
 
 <body
-  class="lang-{$iso_user}{if $lang_is_rtl} lang-rtl{/if} {$smarty.get.controller|escape|strtolower}{if $collapse_menu} page-sidebar-closed{/if}{if isset($is_multishop) && $is_multishop} multishop-enabled{/if}{if isset($lite_display) && $lite_display} light_display_layout{/if}"
+  class="lang-{$iso_user}{if $lang_is_rtl} lang-rtl{/if} {$controller_name|escape|strtolower}{if $collapse_menu} page-sidebar-closed{/if}{if isset($is_multishop) && $is_multishop} multishop-enabled{/if}{if isset($lite_display) && $lite_display} light_display_layout{/if}"
```

Comment:
Will this change keep working with BO pages not migrated yet ?

Reply:
No idea, looks like magic :/

Reply:
Yes, this keeps working with BO pages not migrated because I had this in all Helper!

Reply:
Pages not migrated yet don't use the new theme's layout (they use `default`). The original GET parameter "controller" was always available in legacy controllers because it's what the dispatcher used to tell which controller it had to route to. With symfony routing, this parameter has become obsolete, but it was kept nonetheless in AdminLegacyLayoutController: https://github.com/PrestaShop/PrestaShop/blob/1.7.8.0/controllers/admin/AdminLegacyLayoutController.php#L114

In order to avoid a regression in "vertical" pages which use the legacy layout, I think that the line I mentioned above should be removed, and replaced by the injection of `controller_name` in AdminLegacyLayoutController::initContent().

Reply:
@tegbessou I don't remember the details because it's been so long, but I think that we should remove the assignment to $_GET['controller'] from [this line in AdminLegacyLayoutController]( https://github.com/PrestaShop/PrestaShop/blob/1.7.8.0/controllers/admin/AdminLegacyLayoutController.php?rgh-link-date=2022-03-02T14%3A50%3A01Z#L114) because the parameter is not needed anymore in migrated pages (it is used by the legacy Dispatcher to choose the controller to execute, so no longer needed in Symfony pages).

If we do remove that assignment, $_GET['controller'] becomes empty. This will break the assignment of 'controller_name' in [AdminController::initHeader()](https://github.com/PrestaShop/PrestaShop/blob/1.7.8.0/classes/controller/AdminController.php#L2063), so that line must be changed to use `$this->controller_name` instead of `Tools::getValue('controller')`.